src/explore-education-statistics-admin/src/pages/release/content/components/EditableKeyStatText.tsx

```diff
@@ -25,21 +25,25 @@ export default function EditableKeyStatText({
   onRemove,
   onSubmit,
 }: EditableKeyStatTextProps) {
+  const [keyStatisticId, setKeyStatisticId] = useState('');
   const [showForm, toggleShowForm] = useToggle(false);
 
   const handleSubmit = useCallback(
     async (values: KeyStatTextFormValues) => {
       await onSubmit(values);
+      setKeyStatisticId('');
       toggleShowForm.off();
     },
-    [onSubmit, toggleShowForm],
+    [onSubmit, setKeyStatisticId, toggleShowForm],
   );
 
   if (showForm) {
     return (
       <EditableKeyStatTextForm
         keyStat={keyStat}
-        keyStatisticGuidanceTitles={keyStatisticGuidanceTitles}
+        keyStatisticGuidanceTitles={keyStatisticGuidanceTitles?.filter(
+          keyStatTitle => keyStatTitle === keyStatisticId,
```

Comment:
I don't think this will be working currently, as we aren't setting `keyStatisticId` to the keyStat's title.  

We're wanting to remove the title of _this_ keyStat form the list of all keyStatisticGuidanceTitles, we can use `keyState.title` in the filter. eg:
```js
keyStatisticGuidanceTitles?.filter(
          keyStatTitle => keyStatTitle === keyStat.title)
```